### PR TITLE
✨ Send available condition events for managed cluster

### PR DESCRIPTION
**Pull request by zhujian7**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/registration/hub/lease/controller.go [hunk 1]
```diff
@@ -150,6 +155,10 @@ func (c *leaseController) updateClusterStatus(ctx context.Context, cluster *clus
 		c.eventRecorder.Eventf("ManagedClusterAvailableConditionUpdated",
 			"update managed cluster %q available condition to unknown, due to its lease is not updated constantly",
 			cluster.Name)
+		newClusterCopy := newCluster.DeepCopy()
```
[the remark this answers is not in the document]
Comment (zhujian7) [reply pkg/registration/hub/lease/controller.go:158]: I am not sure, these two events are sent to different namespaces.

File: pkg/registration/spoke/managedcluster/status_controller.go [hunk 2]
```diff
@@ -112,9 +121,39 @@ func (c *managedClusterStatusController) sync(ctx context.Context, syncCtx facto
...
+	newCondition := meta.FindStatusCondition(newCluster.Status.Conditions, clusterv1.ManagedClusterConditionAvailable)
+	if newCondition == nil {
+		return
+	}
+	if condition != nil && condition.Status == newCondition.Status {
+
+		return
+	}
+
+	newClusterCopy := newCluster.DeepCopy()
```
[the remark this answers is not in the document]
Comment (zhujian7) [reply pkg/registration/spoke/managedcluster/status_controller.go:146]: we need to set namespace for the managedcluster for sending event to the managedcluster ns, so I copied here.

File: pkg/registration/spoke/spokeagent.go [hunk 3]
```diff
@@ -371,6 +373,10 @@ func (o *SpokeAgentConfig) RunSpokeAgentWithSpokeInformers(ctx context.Context,
 		recorder,
 	)
 
+	hubEventRecorder, err := helpers.NewEventRecorder(ctx, clusterscheme.Scheme, hubKubeClient, "registrationAgent")
```
[the remark this answers is not in the document]
Comment (zhujian7) [reply pkg/registration/spoke/spokeagent.go:376]: do not quite understand, you mean the "registrationAgent" should be "klusterlet"?

[the remark this answers is not in the document]
Comment (qiujian16) [reply pkg/registration/spoke/spokeagent.go:376]: yes

File: pkg/registration/hub/lease/controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (qiujian16) [reply pkg/registration/hub/lease/controller.go:158]: I think we do not need to send to open-cluster-management-hub again.

File: pkg/registration/spoke/managedcluster/status_controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (zhujian7) [reply pkg/registration/spoke/managedcluster/status_controller.go]: the newCluster is already copied previously, so deepCopy here is not needed, fixed.

File: pkg/registration/spoke/spokeagent.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhujian7) [reply pkg/registration/spoke/spokeagent.go]: changed the component name to klusterlet-agent

**Pull request merged**